cmd/kubeadm/app/phases/certs/certs_test.go

```diff
@@ -364,7 +364,7 @@ func TestWriteKeyFilesIfNotExist(t *testing.T) {
 		}
 
 		//TODO: check if there is a better method to compare keys
-		if resultingKey.D == key.D {
+		if resultingKey.D == test.expectedKey.D {
```

Comment:
I think, that we need a check for `test.expectedKey != nil`, because there is a test case where `expectedKey` is not initialized (left to nil).

Reply:
This code path is unreachable when an error is expected and `test.expectedKey` is `nil` (though it'd be safer to check `test.expectedKey` instead of `expectedErr` indeed).

But... after looking at this code a couple of minutes longer I don't understand what it actually checks: it compares private keys and in case of their equality reports an error about the keys not matching. I'd expect an opposite.

Reply:
Oops, the PR got merged already, but I've addressed the comment in https://github.com/kubernetes/kubernetes/pull/76390.